vbfml/scripts/preprocess_image.py

```diff
@@ -0,0 +1,431 @@
+    met = {
+        "JetImage_pixels": np.zeros(number_event),
+        "JetImage_pixels_preprocessed": np.zeros(number_event),
+    }
+
+    # iterate by batches of 500 over the tree, aim to save memoryquit
+    for sub_tree in tree.iterate(
+        [
+            "JetImage_pixels_preprocessed",
+            "JetImage_pixels",
+        ],
+        entry_start=start,
+        entry_stop=stop,
+        step_size=batch_size,
+        library="np",
+    ):
+
+        batch_counter += 1
+        print(f"batch {batch_counter}/{batch_number}")
+
+        batch_index = (batch_counter - 1) * batch_size
+
+        # fix size of last batch
+        if (batch_counter == batch_number) and (number_event % batch_size != 0):
+            batch_size_iterate = number_event % batch_size
+        else:
+            batch_size_iterate = batch_size
+
+        # check the format of the array to be 'uint8'
+        if sub_tree["JetImage_pixels_preprocessed"].dtype != "uint8":
+            print(
+                f"Warning ! event {(batch_counter-1)*batch_size + j+1} doesn't have format uint8"
+            )
+
+        for j in range(batch_size_iterate):
+
+            a, b = 0, 0
+
+            for label in met:
+                image = sub_tree[label][j].reshape(im_shape)
+                energy, energy_x, energy_y = 0, 0, 0
+
+                # compute the MET from the image
+                for i in range(n_eta_bins):
+                    for k in range(n_phi_bins):
+                        energy_x += (
+                            image[i][k]
+                            * np.cos(phi_centers[k])
+                            / np.cosh(eta_centers[i])
+                        )
+                        energy_y += (
+                            image[i][k]
+                            * np.sin(phi_centers[k])
+                            / np.cosh(eta_centers[i])
+                        )
+
+                energy = np.sqrt(np.square(energy_x) + np.square(energy_y))
+
+                if label == "JetImage_pixels":
+                    a = energy
+                else:
+                    b = energy
+
+                met[label][(batch_counter - 1) * batch_size + j] = energy
+
+            if (a == 0) or (b == 0):
+                print(
+                    f"Event {(batch_counter-1)*batch_size + j+1} of file has 0 MET (empty picture maybe)"
+                )
+
+            if round(a, 6) != round(b, 6):
+                print(
+                    f"event {(batch_counter-1)*batch_size + j+1} does not match ({a},{b} )"
+                )
+
+    with open(cache, "wb") as f:
+        pickle.dump(met, f)
+
+
+@cli.command()
+@click.option(
+    "-i",
+    "--input-dir",
+    required=True,
+    help="Path to the directory with the pkl files",
+)
+@click.option(
+    "-n",
+    "--name-save",
+    default="MET_distribution",
+    required=False,
+    help="Name of the directory where the plots are saved",
+)
+def met_dist(input_dir: str, name_save: str):
```

Comment:
Is this function used to plot the MET distributions for processed and unprocessed, side by side? Could we maybe rename this to something like `plot_met` (also following the earlier naming convention of `check_met` )? And could you add a short docstring under the function definition to explain what this function is doing?